org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/framework/android/AndroidSupport.java

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Microsoft Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+package org.eclipse.jdt.ls.core.internal.framework.android;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import org.eclipse.buildship.core.GradleBuild;
+import org.eclipse.buildship.core.GradleCore;
+import org.eclipse.buildship.core.internal.util.gradle.GradleVersion;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.ls.core.internal.JavaClientConnection.JavaLanguageClient;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.ProgressReport;
+import org.eclipse.jdt.ls.core.internal.ProjectUtils;
+import org.eclipse.jdt.ls.core.internal.framework.IFrameworkSupport;
+import org.eclipse.jdt.ls.core.internal.preferences.PreferenceManager;
+import org.gradle.tooling.model.GradleProject;
+import org.gradle.tooling.model.GradleTask;
+import org.gradle.tooling.model.build.BuildEnvironment;
+import org.gradle.tooling.model.build.GradleEnvironment;
+
+public class AndroidSupport implements IFrameworkSupport {
+
+	public static String SYNCHRONIZATION_TASKS_API_GRADLE = "5.4";
+	public static String ASSEMBLE_TASK_NAME = "assemble";
+
+	@Override
+	public void onDidProjectsImported(IProgressMonitor monitor) {
+		PreferenceManager preferencesManager = JavaLanguageServerPlugin.getPreferencesManager();
+		if (preferencesManager == null) {
+			return;
+		}
+
+		if (!preferencesManager.getPreferences().isAndroidSupportEnabled()) {
+			return;
+		}
+
+		List<IProject> projects = ProjectUtils.getGradleProjects();
+		if (projects.isEmpty()) {
+			return;
+		}
+
+		JavaLanguageClient client = JavaLanguageServerPlugin.getProjectsManager().getConnection();
+		ProgressReport progressReport = new ProgressReport(UUID.randomUUID().toString());
+		progressReport.setTask("Running Gradle tasks");
+		progressReport.setComplete(false);
+		progressReport.setTotalWork(projects.size());
```

Comment:
Sicne the workDone is not set increasingly. You can have two choice here:

- increase it after each round of the for loop
- Just set the total work to 1.

Reply:
Let's set totol work to 1 since this for loop has a lot of "continue"s and it would make the code complex if we set multiple `workDone`s